components/kyma-environment-broker/internal/process/update/initialisation_step.go

```diff
@@ -79,6 +84,31 @@ func (s *InitialisationStep) Run(operation internal.UpdatingOperation, log logru
 		return s.operationManager.OperationSucceeded(operation, fmt.Sprintf("operation preempted by deprovisioning %s", lastOp.ID), log)
 	}
 
+	var version *internal.RuntimeVersionData
+	if operation.RuntimeVersion.IsEmpty() {
+		version, err = s.runtimeVerConfigurator.ForUpdating(operation)
+		if err != nil {
+			return s.operationManager.RetryOperation(operation, err.Error(), 5*time.Second, 1*time.Minute, log)
+		}
+	}
+	var lrs internal.RuntimeState
+	if version.MajorVersion == 2 {
+		lrs, err = s.runtimeStatesDb.GetLatestWithReconcilerInputByRuntimeID(operation.RuntimeID)
+		if err != nil {
+			return s.operationManager.RetryOperation(operation, err.Error(), 5*time.Second, 1*time.Minute, log)
+		}
+	}
+	op, delay := s.operationManager.UpdateOperation(operation, func(op *internal.UpdatingOperation) {
+		if version != nil {
+			op.RuntimeVersion = *version
+		}
+		op.LastRuntimeState = lrs
+	}, log)
+	operation = op
+	if delay != 0 {
+		return operation, delay, nil
+	}
+
```

Comment:
this code block which sets last runtime state sets it as an empty (zero value) value for kyma 1. If someone will implement to use it for kyma one then we won't see any error directly. let's discuss it

Reply:
Thanks! good to know, I will introduce a new step for the "runtime" stage, I think that is the only stage requiring this information.